src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { SortType } from './types/sortType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [query, setQuery] = useState('');
+  const [sortType, setSortType] = useState<SortType | string>(SortType.ALL);
+
+  const loadedUserFromServer = useCallback(async () => {
+    try {
+      const UserFromServer = await getTodos();
+
+      setTodos(UserFromServer);
+      setIsLoaded(true);
+    } catch (error) {
+      throw new Error('Error. Can`t upload todos');
+    }
+  }, []);
+
+  useEffect(() => {
+    loadedUserFromServer();
+  }, []);
+
+  const filterTodos = () => {
+    const sortedTodos = todos.filter(todo => {
+      switch (sortType) {
+        case SortType.ACTIVE:
+          return !todo.completed;
+        case SortType.COMPLETED:
+          return todo.completed;
+        default:
+          return todo;
+      }
+    });
+
+    return sortedTodos.filter(todo => (
+      todo.title.toLowerCase().includes(query.toLowerCase())
```

Comment:
will be better to create separate variable for storing query in lower case before calling filter method